### PR TITLE
W-14557165: Updated User flows for overriding security configuration through file

### PR DESCRIPTION
**Pull request by arpitg-1**
… ssh connection

# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

**Review by dipeshkumardutta-sfemu — DISMISSED** [review]
LGTM

File: sftp/2.1/modules/ROOT/pages/sftp-connector-upgrade-guide.adoc [hunk 1]
```diff
@@ -154,14 +154,47 @@ For details, review the following table:
...
+|`diffie-hellman-group14-sha1` (Deprecated)|`rsa-sha2-256`|
+|`diffie-hellman-group-exchange-sha1` (Deprecated)|`ssh-rsa`|
+||`ssh-dss` (Deprecated)|
+||`ssh-dss-cert-v01@openssh.com` (Deprecated)|
+||`ssh-rsa-cert-v01@openssh.com` (Deprecated)|
+|===
+
+[NOTE]
+* These algorithms require Bouncy Castle on Java versions before Java 11.
+* Provide externally EdDSA (Edwards-curve Digital Signature Algorithm)library.
```
Comment (arpitg-1) [inline sftp/2.1/modules/ROOT/pages/sftp-connector-upgrade-guide.adoc]: Hi @kevintroller , actually, double * was deliberately added as in the above table, there are few algos which have * after them and some have ** after them. So, just wanted to ensure that point highlighted with single * inside NOTE corresponds to those algos which have single * only.

If you have any other way to better represent, please check.

File: sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc [hunk 2]
```diff
@@ -0,0 +1,64 @@
+= Configure Security Parameters of SSH Connection
+
+ The following example shows how to configure the Anypoint Connector for SFTP (SFTP Connector) to override security parameters using Anypoint Studio.
```
Comment (sarathecoubian) [inline sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc]:
```suggestion
The following example shows how to configure the Anypoint Connector for SFTP (SFTP Connector) to override security parameters using Anypoint Studio.
```

Comment (sarathecoubian) [reply sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc]: (created a formatting error)

File: sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+----
+KexAlgorithms ecdh-sha2-nistp521,ecdh-sha2-nistp384,ecdh-sha2-nistp256,diffie-hellman-group18-sha512,diffie-hellman-group17-sha512,diffie-hellman-group16-sha512,diffie-hellman-group14-sha256
+Ciphers aes128-cbc,aes128-ctr,aes192-cbc,aes192-ctr,aes256-cbc,aes256-ctr
+HostKeyAlgorithms ssh-rsa,rsa-sha2-256,rsa-sha2-512
+MACs hmac-sha1,hmac-sha2-256,hmac-sha2-512
+----
+image::sftp-override-config-file-contents.png[Contents of the override config file]
+
+[start=3]
+. Configure the keys and values accordingly. If you do not provide any key, then default values are taken into consideration.
```
Comment (sarathecoubian) [inline sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc]:
```suggestion
. Configure the keys and values accordingly. If you do not provide a key, then default values are used.
```

Comment (sarathecoubian) [reply sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc]: I would try to change "accordingly" to "according to [whatever it is dependent on]" if you know what that means. For the second part, I am not sure what you mean by taken into consideration. Does it evaluate them? Might say "evaluated and used."

File: sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+The following example shows where the config file is present in  the`src/main/resources` folder:
```
Comment (sarathecoubian) [inline sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc]:
```suggestion
The following example shows the location of the config file within  the `src/main/resources` folder:
```

File: sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+== Configure the Override File in Anypoint Studio
+
+Follow these steps to configure the override file:
+
+. In Studio, navigate to the *Global Elements* tab.
+. In the Choose Global Type window, type `sftp` in the search bar.
+. Select *SFTP Config*.
+. Click *OK*.
+. Go to *Advanced* Tab.
```
Comment (sarathecoubian) [inline sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc]:
```suggestion
. Go to the *Advanced* Tab.
```

File: sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+. Under *Security Configuration* provide `SSH Config Override File`.
```
Comment (sarathecoubian) [inline sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc]:
```suggestion
. Under *Security Configuration*, provide `SSH Config Override File`.
```

Comment (sarathecoubian) [reply sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc]: Consider select or enter instead.

File: sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+. Provide the path to the file. Example: `${app.home}/mule_sshd_config`.
```
Comment (sarathecoubian) [inline sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc]:
```suggestion
. Enter the path to the file. Example: `${app.home}/mule_sshd_config`.
```

File: sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    <sftp:connection host="localhost" sshConfigOverride="${app.home}/mule_sshd_config">
+    </sftp:connection>
+</sftp:config>
+----
+
+== Verify the Override File Creation
+
+Consider the following points when verifying the file creation:
+
+*  Search for the log `Read the config file` and verify that the filename and the configuration values fetched from the file are valid.
```
Comment (sarathecoubian) [inline sftp/2.1/modules/ROOT/pages/sftp-security-configuration-override.adoc]:
```suggestion
* Search for the log `Read the config file` and verify that the filename and the configuration values fetched from the file are valid.
```

**Review by sarathecoubian — APPROVED** [review]

**Pull request merged**